### PR TITLE
Feature/914 application form tweaks

### PR DESCRIPTION
**Pull request by ramonh**
Related issue #914 and private issues

This aims to prepare the application form for the 2018 summer of code:

- Add requirement for university students to clarify their courses and times
- Add character limit (same as the others) to the code sample field
- Clarify that code sample field should not include copy-pasted code but rather should be links
- Hide the gender self-identification field from teammates
- Remove diversity field requirement
- Remove warnings related to the blind selection process (as this will no longer take place)

### CODE REVIEW

**Comment by nerdyvaishali** [comment]
Hi @ramonh... All changes looks good to me. Thanks for working on them.

File: app/helpers/application_drafts_helper.rb [hunk 1]
```diff
@@ -18,6 +18,12 @@ module ApplicationDraftsHelper
                     :application_money,
                     :application_minimum_money]
 
+  def in_same_team_as_other_student?(student)
```
Comment (klappradla) [inline app/helpers/application_drafts_helper.rb]: Hm, just wondering, can't you reuse `private_application_data` from the `ApplicationDraftsHelper` for that 🤔 

If I understand your motivations here, it should basically be the same...

File: app/views/application_drafts/_student_fields.html.slim [hunk 2]
```diff
@@ -9,17 +9,15 @@
...
-  = s.input :application_diversity, as: :character_limited, label: 'Tell us about any special circumstances which you want us to consider while reviewing your application', input_html: { disabled: !may_edit?(s.object) }, wrapper_html: { class: student_field_error_class(index, application_draft, :application_diversity) }, hint: (student_field_error_hint(index, application_draft, :application_diversity) || 'E.g., LGBTQI, people of color, people with disabilities, single parents, people taking care of family members, etc. Please do not mention your name, age and location here.')
+  - unless in_same_team_as_other_student?(student)
```
Comment (klappradla) [inline app/views/application_drafts/_student_fields.html.slim]: See my comment in the helper file above - I think this problem is already solved elsewhere.

File: app/views/application_drafts/_student_fields.html.slim [hunk 3]
```diff
@@ -9,17 +9,15 @@
...
-
   h2 Background Information
 
-  = s.input :application_about, as: :character_limited, label: 'Please tell us a little bit about yourself. What is your background?', input_html: { rows: 5, disabled: !may_edit?(s.object) }, wrapper_html: { class: student_field_error_class(index, application_draft, :application_about)}, hint: (student_field_error_hint(index, application_draft, :application_about) || 'Only share what you feel comfortable with.')
+  = s.input :application_about, as: :character_limited, label: "Please tell us a little bit about yourself. What is your background?<br>If you're a university student, please tell us what courses you'll be taking at any time during the summer, as well as how many weekly hours these take up.".html_safe, input_html: { rows: 5, disabled: !may_edit?(s.object) }, wrapper_html: { class: student_field_error_class(index, application_draft, :application_about)}, hint: (student_field_error_hint(index, application_draft, :application_about) || 'Only share what you feel comfortable with.')
```
Comment (klappradla) [inline app/views/application_drafts/_student_fields.html.slim]: Can you please move these changes to the label into the `config/locales/views.en.yml` file I created in #765 when rebasing this on `master`?

File: app/views/application_drafts/_student_fields.html.slim [hunk 4]
```diff
@@ -9,17 +9,15 @@
...
+  - unless in_same_team_as_other_student?(student)
+    = s.input :application_gender_identification, as: :string, label: 'Your gender self-identification', input_html: { disabled: !may_edit?(s.object), "data-genders-src" => ApplicationDraft::GENDER_LIST.to_json, class: "gender-form-field"}, wrapper_html: { class: student_field_error_class(index, application_draft, :application_gender_identification) }, hint: (student_field_error_hint(index, application_draft, :application_gender_identification) ||  'No-one is excluded from applying on the basis of gender, but people who self-identify as female, or have experiences being socialized as female, are given preference during selection.')
```
Comment (klappradla) [inline app/views/application_drafts/_student_fields.html.slim]: When rebasing this to `master` please be aware that I extracted all the labels into translation files.

File: app/views/application_drafts/_student_fields.html.slim [hunk 5]
```diff
@@ -9,17 +9,15 @@
...
+  = s.input :application_diversity, as: :character_limited, label: 'Tell us about any special circumstances which you want us to consider while reviewing your application', input_html: { disabled: !may_edit?(s.object) }, wrapper_html: { class: student_field_error_class(index, application_draft, :application_diversity) }, hint: (student_field_error_hint(index, application_draft, :application_diversity) || 'E.g., LGBTQI, people of color, people with disabilities, single parents, people taking care of family members, etc.')
```
Comment (klappradla) [inline app/views/application_drafts/_student_fields.html.slim]: Same here: when rebasing this to `master` please be aware that I extracted all the labels into translation files.

File: app/views/application_drafts/_student_fields.html.slim [hunk 6]
```diff
@@ -55,7 +53,7 @@
...
+  = s.input :application_code_samples, as: :character_limited, label: "Please provide links to examples as well as short explanations of your code, preferably in the language of the project you're applying for", input_html: { rows: 4, disabled: !may_edit?(s.object) }, wrapper_html: { class: student_field_error_class(index, application_draft, :application_code_samples) }, hint: (student_field_error_hint(index, application_draft, :application_code_samples) || "E.g., links to completed coding challenge provided by the mentor (if present), commits or pull requests on Github or other concise examples of your code. Links to GitHub profiles and repositories won't be considered. Please do not directly copy-paste code here.")
```
Comment (klappradla) [inline app/views/application_drafts/_student_fields.html.slim]: Same here: When rebasing this to `master` please be aware that I extracted all the labels into translation files.

File: app/helpers/application_drafts_helper.rb (hunk 1 above)
Comment (ramonh) [reply app/helpers/application_drafts_helper.rb]: Oh, hm, looks like yes most probs. Thanks for the tip 😅

**Comment by ramonh** [comment]
Hey @klappradla, thanks for the feedback! 

Phew! I survived the rebase 🙈 

How's it looking?

**Comment by klappradla** [comment]
Yay, looks good! Great job on the rebase hell - sorry again 😹 

Let's ⛵️

**Pull request merged**